docs/content/v0-upgrade.md

```diff
@@ -0,0 +1,464 @@
+body and no value assignment, i.e. a solitary reference, are not allowed in
+the v1.0 Rego syntax:
+
+| rule  | output in v0              | output in v1.0 |
+| ----- | ------------------------- | -------------- |
+| p     | compile error             | compile error  |
+| p.a   | {“p”: {“a”}}              | compile error  |
+| p.a.b | {“p”: {“a”: {“b”: true}}} | compile error  |
+
+The below table gives examples of v0 valid Rego syntax which are
+invalid in OPA v1.0, along with the equivalent valid syntax in OPA v1.0:
+
+| invalid in v1.0 | v1.0 equivalent            | Note                    |
+| --------------- | -------------------------- | ----------------------- |
+| p { true }      | p if { true }              | Single-value rule       |
+| p.a             | p contains "a"             | Multi-value insertion   |
+| p.a { true }    | p contains "a" if { true } | Multi-value rule        |
+| p.a.b           | p.a.b := true              | Single-value assignment |
+| p.a.b { true }  | p.a.b if { true }          | Single-value rule       |
+
+Following is an example of how to define a rule that generates a set:
+
+```rego
+package play
+
+a contains b if { b := 1 }
+```
+
+When the above rule is evaluated the output is (sets are serialized into arrays
+in JSON):
+
+```json
+{
+  "a": [1]
+}
+```
+
+Following is an example of how to define a rule that generates an object:
+
+```rego
+package play
+
+a[b] if { b := 1}
+```
+
+When the above rule is evaluated the output is:
+
+```json
+{
+  "a": {
+    "1": true
+  }
+}
+```
+
+The requirement of `if` and `contains` keywords remove the ambiguity
+between single-value and multi-value rule declaration. This makes Rego code
+easier to author and read; thereby making it simpler for users to author their
+policies.
+
+### Prohibit duplicate imports
+
+The Rego compiler supports [strict mode](../policy-language/#strict-mode) which
+enforces additional constraints and safety checks during compilation. One of
+these checks is to prohibit duplicate imports where one import shadows another.
+OPA v1.0 enforces this check by default.
+
+An import shadowing another is most likely an authoring error and probably
+unintentional. OPA checking this by default will help to avoid policy
+evaluations resulting in error-prone decisions.
+
+### `input` and `data` keywords are reserved
+
+The Rego compiler supports [strict mode](../policy-language/#strict-mode), which
+enforces additional constraints and safety checks during compilation. One of
+these checks is to ensure that `input` and `data` are reserved keywords and may
+not be used as names for rules and variable assignments.
+
+The `input` document holds the user-provided input, while the data pushed into
+OPA and rule evaluation results are nested under the `data` document. Hence, if
+a rule or variable shadows `input` or `data` you have the unintended consequence
+of erasing information under these, resulting in incorrect policy decisions. In
```

Comment:
```suggestion
a rule or variable shadows `input` or `data` you have the unintended consequence
of erasing information under these inside the local scope, resulting in incorrect policy decisions. In
```

These documents are immutable, but the old text reads like this is a global mutation.

Reply:
Updated